Malicious content causes `verifiable.ParseCredential()` to panic

**What I'm trying to do**

I'm implementing conformance testing for https://github.com/w3c-ccg/traceability-interop, and our implementation uses aries-framework-go on the back end. Some of the negative testing inputs for credential verification cause the aries code to panic.

**Expected result**

I'm expecting malformed input to result in some kind of parse error.

**Actual result**

Actual result with maliciously crafted content is a panic caused by an [unguarded type assertion](https://github.com/hyperledger/aries-framework-go/blob/main/pkg/doc/signature/proof/proof.go#L165-L171).

The following is an example of the panic output if a vc with `proof.created` is set to an integer:

```
panic: interface conversion: interface {} is float64, not string

goroutine 1 [running]:
github.com/hyperledger/aries-framework-go/pkg/doc/signature/proof.stringEntry(...)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/signature/proof/proof.go:170
github.com/hyperledger/aries-framework-go/pkg/doc/signature/proof.NewProof(0x1546300)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/signature/proof/proof.go:64 +0x905
github.com/hyperledger/aries-framework-go/pkg/doc/signature/proof.GetProofs(0x90)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/signature/proof/utils.go:41 +0x1ad
github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier.(*DocumentVerifier).verifyObject(0xc00029fcb8, 0xc00013a5a8, {0xc0002eb430, 0x2, 0x2})
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/signature/verifier/verifier.go:83 +0x56
github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier.(*DocumentVerifier).Verify(0xc00029fe18, {0xc0002782c0, 0x2b2, 0x2b2}, {0xc0002eb430, 0x2, 0x2})
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/signature/verifier/verifier.go:78 +0xda
github.com/hyperledger/aries-framework-go/pkg/doc/verifiable.checkLinkedDataProof({0xc0002782c0, 0x2b2, 0x2b2}, {0xc0002eb420, 0xc00013a598, 0x1}, 0xc0002eb220, 0xc000278580)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/verifiable/linked_data_proof.go:78 +0x1a6
github.com/hyperledger/aries-framework-go/pkg/doc/verifiable.checkEmbeddedProof({0xc0002782c0, 0x2b2, 0x2b2}, 0xc00029fdf0)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/verifiable/embedded_proof.go:100 +0x2d1
github.com/hyperledger/aries-framework-go/pkg/doc/verifiable.decodeRaw({0xc0002782c0, 0x2b2, 0x2b2}, 0xc000156240)
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/verifiable/credential.go:1072 +0x349
github.com/hyperledger/aries-framework-go/pkg/doc/verifiable.ParseCredential({0xc0002782c0, 0x2b2, 0x2b2}, {0xc00029ff48, 0xc0000001a0, 0xc000056738})
        /example/src/go/pkg/mod/github.com/hyperledger/aries-framework-go@v0.1.9-0.20220921220826-52d9990bb684/pkg/doc/verifiable/credential.go:800 +0x6b
main.main()
        /example/src/panictest/main.go:37 +0x1db
exit status 2
```

**Sample code / test case**

```golang
package main

import (
	"net/http"

	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
	"github.com/hyperledger/aries-framework-go/pkg/vdr"
	"github.com/piprate/json-gold/ld"
)

const vc string = `{
	"@context": ["https://www.w3.org/2018/credentials/v1"],
	"credentialSubject": {"id": "did:example:123"},
	"issuanceDate": "2006-01-02T15:04:05Z",
	"issuer": "did:key:z6MkgVHZNqLBqoQAoGxRiSJP5gLgVEDCJJzT5ZsGEabKtfyn",
	"type": ["VerifiableCredential"],
	"proof": {
		"created": 123,
		"verificationMethod": "did:key:z6MkgVHZNqLBqoQAoGxRiSJP5gLgVEDCJJzT5ZsGEabKtfyn#z6MkgVHZNqLBqoQAoGxRiSJP5gLgVEDCJJzT5ZsGEabKtfyn",
		"proofPurpose": "assertionMethod",
		"type": "Ed25519Signature2018",
		"jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..Csadnd3P2WqJLxAeXMkyx4nkvXvSEssOQriVuwYK0_6zyssqkkuAcKpkUFh8rf4J5JVpng9yvb_2263dLMLZCw"
	}
}`

func main() {
	keyResolver := verifiable.NewVDRKeyResolver(vdr.New())
	publicKeyFetcher := keyResolver.PublicKeyFetcher()
	documentLoader := ld.NewDefaultDocumentLoader(http.DefaultClient)

	verifiable.ParseCredential([]byte(vc),
		verifiable.WithPublicKeyFetcher(publicKeyFetcher),
		verifiable.WithJSONLDDocumentLoader(documentLoader),
	)
}
```

--- a/pkg/doc/signature/proof/proof.go
+++ b/pkg/doc/signature/proof/proof.go
@@ -167,7 +167,11 @@ func stringEntry(entry interface{}) string {
 		return ""
 	}
 
-	return entry.(string)
+	if strVal, ok := entry.(string); ok {
+		return strVal
+	}
+
+	return ""
 }
 
 // JSONLdObject returns map that represents JSON LD Object.

--- a/pkg/doc/signature/proof/proof_test.go
+++ b/pkg/doc/signature/proof/proof_test.go
@@ -155,6 +155,183 @@ func TestProof(t *testing.T) {
 	})
 }
 
+func TestInvalidInterfaceTypeShouldNotPanic(t *testing.T) {
+	t.Run("does not panic if type is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               123,
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if created is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            123,
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+		})
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "parsing time")
+	})
+
+	t.Run("does not panic if creator is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            123,
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if verificationMethod is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": 123,
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if proofValue is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         123,
+		})
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "signature is not defined")
+	})
+
+	t.Run("does not panic if jws is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"jws":                123,
+		})
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "signature is not defined")
+	})
+
+	t.Run("does not panic if proofPurpose is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+			"proofPurpose":       123,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if domain is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             123,
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if nonce is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              123,
+			"proofValue":         proofValueBase64,
+		})
+		require.NoError(t, err)
+	})
+
+	t.Run("does not panic if challenge is not a string", func(t *testing.T) {
+		defer func() {
+			r := recover()
+			require.Nil(t, r)
+		}()
+		_, err := NewProof(map[string]interface{}{
+			"type":               "type",
+			"creator":            "didID",
+			"verificationMethod": "did:example:123456#key1",
+			"created":            "2018-03-15T00:00:00Z",
+			"domain":             "abc.com",
+			"nonce":              "",
+			"proofValue":         proofValueBase64,
+			"challenge":          123,
+		})
+		require.NoError(t, err)
+	})
+}
+
 func TestInvalidProofValue(t *testing.T) {
 	// invalid proof value
 	p, err := NewProof(map[string]interface{}{

